### PR TITLE
Added icon to both capability and talent

### LINKED ISSUE #945: [FEATURE] Add support in Contributions BB to upload an icon for Talents and Capabilities
1. Look at Content BB POST /image endpoint
2. Add a URL field to Talent and Capability schemas to store the icon URL

### PR DESCRIPTION
**Pull request by ywkim312**
## Description
Added icon field to both capability and talent. You can test using the endpoint directly, NOT going through catalog, by just adding icon field in capability and talent json in contribution json

**Fixes #945 **

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Immediately
- [ ] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by minump** [comment]
Screen Shot 2022-06-23 at 11 12 08 AM
user-images.githubusercontent.com/22112888/175346160-de610648-0f37-421d-aa5f-d6bd9a8e099a.png)

On posting a capability with "icon" field, it is not being reflected in robo3T. Am I missing something here?

**Comment by minump** [comment]
Screen Shot 2022-06-23 at 11 12 08 AM

Screen Shot 2022-06-23 at 11 12 32 AM

On posting a capability with a "icon" field, it is not present in robo3T. Am I missing something here?

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 ### Added
+- Display of Talent Self Certification fields when empty. [#913](https://github.com/rokwire/rokwire-building-blocks-api/issues/913)
```
Comment (minump) [inline CHANGELOG.md:9]: This issue is there in the Changed section. So maybe delete the line from here?

**Comment by minump** [comment]
> > Screen Shot 2022-06-23 at 11 12 08 AM
> > Screen Shot 2022-06-23 at 11 12 32 AM
> > On posting a capability with a "icon" field, it is not present in robo3T. Am I missing something here?
> 
> It is weird. It should be in both talent and capability. Can you pull the PR and try again? image

@ywkim312 So sorry. There was a typo and hence it didnt reflect. My apologies. I am able to see the field now. Thanks!

**Review by minump — APPROVED** [review]
Tested and working fine. Approved with minor change to changelog

**Comment by sandeep-ps** [comment]
image

I'm seeing an error when posting from Catalog without providing an image icon. I think the icon needs to be made nullable in Contributions BB schema is None is being sent.  Thanks.

**Comment by ywkim312** [comment]
> image
> 
> I'm seeing an error when posting a Capability from Catalog without providing an image icon. I think the icon field needs to be made nullable in Contributions BB schema is `None` is being sent. Thanks.

Added nullable to icons in both capability and talent

**Comment by sandeep-ps** [comment]
Thanks. Will approve and merge.

**Pull request merged**